src/lwt/win/lwt_unix_impl.ml

```diff
@@ -363,6 +369,12 @@ if !d then Format.eprintf "Start writing@.";
 if !d then Format.eprintf "Writing started@.";
   res
 
+let write ch s pos len =
+  _write_string ch (Bytes.to_string s) pos len
```

Comment:
I think it's better let `write` untouched (and take a bytes), and to add a new function `write_substring` whose contents is the same as `write` except for the needed ajustments to make it take a string.

Reply:
The CICD portion took the most mental time in the last couple of weeks, so I'm fuzzy about this choice. But I believe it was a combination of minimizing code duplication (especially for `write` and `write_substring`), seeing relatively easy conversion of Bytes to Strings (`(Bytes.to_string s)`) vs the scary reverse conversion using `unsafe_blit_to_bytes`, and getting the functional interface correct for all the used locations.

I'll will look at it and see if I can refactor it into a form more closely aligned with your suggestion.